linkerd/proxy/client-policy/src/opaq.rs

```diff
@@ -127,6 +127,9 @@ pub(crate) mod proto {
             filters: NO_FILTERS.clone(),
             failure_policy: NonIoErrors::default(),
             distribution,
+            // XXX(eliza): maybe this shouldn't even be present on opaque
+            // routes...might have to turn it into a filter...
```

Comment:
little bit confused about this... I guess if a service is marked as opaque, and it has a route attached to it, the route becomes opaque. Generally, the way I reasoned about things in the past was:

* If service is opaque, we don't do protocol detection
* If endpoint is opaque, we change the target port (but we may do protocol detection)

...so I guess an opaque route is a TCP route in effect? would the request timeout never be a good fit? not sure if my reasoning here is sound 🤷🏻

Reply:
Check out the comments I added in https://github.com/linkerd/linkerd2-proxy/pull/2418/commits/26c9492af1e52ab85b9e1dcbeeed9b791e81dcbd and let me know if that answers your questions?